src/telegraf.ts

```diff
@@ -238,8 +235,16 @@ export class Telegraf<
     return Promise.race([processAll, sleep(this.options.handlerTimeout)])
   }
 
+  private botInfoCall: Promise<tt.UserFromGetMe> | undefined
   async handleUpdate(update: tt.Update, webhookResponse?: http.ServerResponse) {
     debug('Processing update', update.update_id)
+    this.botInfo ??=
+      (debug(
+        'Update',
+        update.update_id,
+        'waiting for `botInfo` to be initialized'
+      ),
```

Comment:
Magic, begone!
```suggestion
```

Reply:
I'm against comma operator. You could do

```ts
if (this.botInfo === undefined && this.botInfoCall === undefined) {
  debug("Initializing `botInfo`...)
}
```

On 2nd thought, this would introduce weird logic duplication... I'd drop the `debug` call for now.

Reply:
> I'm against comma operator.

As in, radically? I feel like logging statements is the only place for them.

Reply:
Yes. But on 3rd thought, this message is useful enough, I'll let it in.